lib/publiccloud/azure.pm

```diff
@@ -59,6 +64,24 @@ sub get_image_id {
     return $self->SUPER::get_image_id($img_url);
 }
 
+=head2 generate_azure_image_definition
+
+    my $definition = $self->generate_azure_image_definition();
+
+Generated the Azure Image name from the job settings. If present, it takes the PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION setting.
+If not present it generated the image definition name based on distri, version, flavor and SKU.
+
+Note: Image definitions needs to be distinct names and can only serve one architecture!
+
+Example: 'SLE-MICRO-5.4-BYOS-AZURE-X86_64-GEN1'
+=cut
+
+sub generate_azure_image_definition {
+    my $sku = get_required_var('ARCH') . '-' . $default_sku;    # PUBLIC_CLOUD_AZURE_SKU is something like 'aarch64-gen2'
```

Comment:
`'aarch64-gen2'` - from where this coming ? I just checked our job group definitions and I failed to find anywhere `PUBLIC_CLOUD_AZURE_SKU=aarch64-gen2`

Reply:
For image definition the `publisher - offer - sku` combination needs to be unique.
That's why it's generated and it does not matter to pass it as variable in this scenario.

On the other hand when we're launching already published images it makes sense to provide those variables.

Reply:
my point that comment is stating that `PUBLIC_CLOUD_AZURE_SKU` can have value like `aarch64-gen2` but I never heard/seen something like that . So or **comment** has false statement than it needs to be removed/updated or I don't know something about `PUBLIC_CLOUD_AZURE_SKU` than I need to update my knowledge

Reply:
Exactly what Pavel says. It comes from the requirement of Azure to identify images via the tuple `(publisher, offer, sku)` and arguably the architecture is best fitted into the sku and not into `publisher` or `offer`.

Reply:
from my knowledge `PUBLIC_CLOUD_AZURE_SKU` can have ONLY two possible values "gen1" or "gen2" . I never have seen architecture inside sku . Am I wrong ?

Reply:
Can you please make a concrete suggestion on how we should address this then?

Reply:
I am not honestly sure. If we leave `SKU` as `gen1|gen2` then the `offer will need to be something like `sle-15-SP5-byos-aarch64`.

Reply:
But the `$sku` here is anyways used only in this function which returns the image definiton name.

Reply:
And also this function (`generate_azure_image_definition`) is not currently used.

Reply:
> And also this function (`generate_azure_image_definition`) is not currently used.

It's used 2 times in `azure.pm` and one time in `provider.pm`.

Reply:
Sorry. My browser is somehow not in shape.